playbooks/ai-on-openlab/pre.yaml

```diff
@@ -0,0 +1,93 @@
+- name: AI On Openlab Pre Job
+  hosts: all
+  become: yes
+  tasks:
+    - name: Set Default Hardware type Parameters
+      set_fact:
+        user_file: "user_data"
+      when: not is_gpu_enabled
+
+    - name: Split Hardware type Parameters for GPU
+      set_fact:
+        user_file: "user_data_gpu"
+      when: is_gpu_enabled
+
+    - name: Shape login Demo User
+      shell:
+        cmd: |
+          set -x
+          set -e
+
+          useradd demo -p 'demo'
+          echo 'demo:demo' | chpasswd
+          usermod -s /bin/bash demo
+          mkdir /home/demo
+          chown demo:demo /home/demo
+          sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config
+          service sshd restart
+        executable: /bin/bash
+
+    - name: Upload Environment Info to Web
+      shell:
+        cmd: |
+          apt install -y python-pip
+          pip install pygithub
+
+          source_file=`ls "{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/" | grep moo-job`
```

Comment:
I think you can just define a variable like: `user_file_path: {{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/`, I found all the `{{ user_file }}` usage is this path

too many redundant variable references ...

Reply:
ok